homeassistant/components/switch/xiaomi_ir.py

```diff
@@ -0,0 +1,196 @@
+"""
+Support for the Xiaomi IR Remote (Chuangmi IR).
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/ir_remote.xiaomi_miio/
+"""
+import asyncio
+import logging
+
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.switch import (
+    DOMAIN, PLATFORM_SCHEMA, SwitchDevice)
+from homeassistant.const import (
+    CONF_COMMAND_OFF, CONF_COMMAND_ON, CONF_FRIENDLY_NAME,
+    CONF_SWITCHES, CONF_HOST, CONF_TOKEN, CONF_SLOT, CONF_TIMEOUT)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util.dt import utcnow
+
+REQUIREMENTS = ['python-miio==0.3.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+SERVICE_LEARN = 'xiaomi_miio_learn_command'
+SERVICE_SEND = 'xiaomi_miio_send_command'
+
+DEFAULT_TIMEOUT = 10
+DEFAULT_SLOT = 1
+
+SWITCH_SCHEMA = vol.Schema({
+    vol.Optional(CONF_COMMAND_OFF, default=None): cv.string,
+    vol.Optional(CONF_COMMAND_ON, default=None): cv.string,
+    vol.Optional(CONF_FRIENDLY_NAME): cv.string,
```

Comment:
Are `on` and `off` the only possible commands? Or should this be done similarly to https://home-assistant.io/components/remote.itach/ with named commands?

Reply:
This feels much better! On and Off really does feel weird when you can send any arbitrary IR command

Reply:
I still think that this should follow the same API as the rest of remotes, there's always `send_command` for any custom commands, if I'm understanding it correctly now.